dartagnan/src/main/java/com/dat3m/dartagnan/expression/ExpressionPrinter.java

```diff
@@ -0,0 +1,98 @@
+package com.dat3m.dartagnan.expression;
+
+import com.dat3m.dartagnan.expression.floats.FloatSizeCast;
+import com.dat3m.dartagnan.expression.floats.IntToFloatCast;
+import com.dat3m.dartagnan.expression.integers.FloatToIntCast;
+import com.dat3m.dartagnan.expression.integers.IntSizeCast;
+import com.dat3m.dartagnan.expression.misc.ConstructExpr;
+import com.dat3m.dartagnan.expression.misc.ExtractExpr;
+import com.dat3m.dartagnan.expression.misc.GEPExpr;
+import com.dat3m.dartagnan.expression.misc.ITEExpr;
+import com.dat3m.dartagnan.program.Register;
+
+import java.util.stream.Collectors;
+
+public final class ExpressionPrinter implements ExpressionVisitor<String> {
+
+    private final boolean qualified;
```

Comment:
This should get a more descriptive name: "qualified" does not in the slightest tell you what is going on here.
Just call it `printRegsWithFuncIds` or similar.